### ISSUE TITLE
Demo App Stuck on Launch on Samsung Galaxy A13 and A11 (Android 13 and 14)

### ISSUE THREAD

**Issue by solveit-aleksei-karmazin** [daily-demos/daily-flutter-demo#16]
### Describe the issue:
I'm a software developer at a company that uses daily_flutter library in our product. We've encountered a device-specific issue with our product and demo app. On Samsung Galaxy A13 and A11 running Android 13 and 14, the app gets stuck on launch. This issue does not occur on iOS devices or other Android devices, so it seems to be specific to these Samsung models.

### Steps to reproduce the issue:

Run the demo app on a Samsung Galaxy A13 or A11 (Android 13/14).
Observe that the app gets stuck on launch.
Expected behavior:
The app should launch and function normally, as it does on other devices.

### Observed behavior:
The app remains stuck on the launch screen and does not proceed further.

### Additional context:

The issue is reproducible only on Samsung Galaxy A13 and A11 devices running Android 13/14.
The app works fine on iOS and other Android devices.
The problem seems to be device-specific.
Environment details:

Library version: daily_flutter 0.22.0
Device models: Samsung Galaxy A13, Samsung Galaxy A11
OS versions: Android 13, Android 14
Any relevant logs or errors: This is logs from Samsung A13 android 14 with demo app launch

[samsung-SM-A135F-Android-14_2024-12-11_120604.logcat.zip](https://github.com/user-attachments/files/18107988/samsung-SM-A135F-Android-14_2024-12-11_120604.logcat.zip)

**Comment by SuryaSharpsell** [comment]
Facing Same issue those mentioned devices and also in few redmi devices

**Comment by SuryaSharpsell** [comment]
Yes, it is happening in latest version of daily_flutter. Flutter version, we are using is 3.24.0. This issue is happening only in those specific devices.

**Comment by filipi87** [comment]
Hi @SuryaSharpsell,

I've tested on a few other Samsung devices but haven’t been able to reproduce the issue yet. I’ll likely need to get access to one of the specific models you mentioned to investigate further.

From the logs you shared, it doesn’t look like the app is even trying to access the camera. When you first install the app, does it request camera permission ?

My current guess is that this might be related to a hardware rendering issue. Flutter 3.24.0 included a fix for a similar problem on Android 14:
- [Release notes – Flutter 3.24.0](https://docs.flutter.dev/release/release-notes/release-notes-3.24.0)
- [GitHub issue #144219](https://github.com/flutter/flutter/issues/144219)
- [GitHub issue #139630](https://github.com/flutter/flutter/issues/139630)

But since you're already using that version, it’s likely something else is going on. I'll continue looking into it.

**Comment by jamsea** [comment]
@filipi87 for context over email we confirmed that the app does request camera permissions and there's no way to proceed without accepting camera permissions

**Comment by filipi87** [comment]
Hi, I found the issue. There was a mismatch in Dart port values between Dart and native code on Android.

This caused undefined behavior on some platforms (especially ARM32), which prevented messages from being sent from the native API to Dart code, causing the app to get stuck at launch.

Tomorrow, I’ll run a few more tests to make sure there’s nothing else we need to fix, but I hope to have a PR ready by then.
